### PR TITLE
grpcdb: close Iterator/ReverseIterator after use

### LINKED ISSUE #3402: Close Iterator in RemoteDB
https://github.com/tendermint/tendermint/blob/e0f8936455029a40287a69d5b0e7baa4d5864da1/libs/db/remotedb/grpcdb/server.go#L139-L161

Though this part of codes does not seem to be used for now, but in case it's used in the future, do we need to close the Iterator in `handleIterator`?

### PR DESCRIPTION
**Pull request by melekes**
Fixes #3402

* [ ] Updated all relevant documentation in docs
* [ ] Updated all code comments where relevant
* [ ] Wrote tests
  Maybe in the future. Code is unused right now.
* [x] Updated CHANGELOG_PENDING.md

### CODE REVIEW

File: libs/db/remotedb/grpcdb/server.go [hunk 1]
```diff
@@ -162,6 +163,7 @@ func (s *server) handleIterator(it db.Iterator, sendFunc func(*protodb.Iterator)
 
 func (s *server) ReverseIterator(query *protodb.Entity, dis protodb.DB_ReverseIteratorServer) error {
 	it := s.db.ReverseIterator(query.Start, query.End)
+	defer it.Close()
 	return s.handleIterator(it, dis.Send)
```
Comment (Liamsi) [inline libs/db/remotedb/grpcdb/server.go]: Should we add a comment to `handleIterator` that the caller is responsible for closing the iter?

Comment (melekes) [reply libs/db/remotedb/grpcdb/server.go]: I think it should be obvious since it's passed as an argument, no?

Comment (Liamsi) [reply libs/db/remotedb/grpcdb/server.go]: Right. I’m just worried that we might forget to close it in other places that will call this (or other) functions.

Comment (melekes) [reply libs/db/remotedb/grpcdb/server.go]: Would be cool to have another program that checks this kind of stuff!

**Pull request merged**